lib/engine/game/g_18_fr/round/share_redemption.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require_relative '../../../round/stock'
+
+module Engine
+  module Game
+    module G18FR
+      module Round
+        class ShareRedemption < Engine::Round::Stock
+          attr_writer :loan_taken, :share_redeemed
+
+          def self.short_name
+            'SRR'
+          end
+
+          def self.round_name
+            'Share Redemption Round'
+          end
+
+          def setup
+            @loan_taken = false
+            @share_redeemed = false
+          end
+
+          def select_entities
+            corporations = @game.corporations.select do |corp|
+              corp.floated? && !corp.share_price.acquisition? && !@game.share_pool.shares_of(corp).empty?
+            end
+
+            if corporations.empty?
+              @log << 'No corporation can redeem shares'
+              finish_round
+            end
+
+            corporations.sort
+          end
+
+          def next_entity!
+            raise GameError, 'Must redeem share if loan taken' if @loan_taken && !@share_redeemed
```

Comment:
Where is @loan_taken and @share_redeemed reset between entities?

Reply:
Good catch!